### ISSUE TITLE
V0.0.1 Release Plan

### ISSUE THREAD

**Issue by cp5555** [microsoft/ltp-megatron-lm#31]
# Release Manager
@yzygitzh 

# Endgame
- [x] Code freeze: May. 26, 2025
- [x] Bug Bash date: May. 27, 2025
- [x] Release date: May 30, 2025

# Framework Features

## Checkpoint - @abuccts
1. - [x] Fix checkpoint convert when using async save (#10)
2. - [x] Support triggering manual GC after checkpoint (#12)
3. - [x] Upload checkpoints to Azure Blob (#19)
4. - [x] Recalculate rampup batch size and data offset (#20)
5. - [x] Support isolated checkpoint saving (#24)

## Dataloader - @yzygitzh 
1. - [x] Improve dataset weighted blending (#15)

## Logging - @yukirora 
1. - [x] Add customized wandb logs (#11)
2. - [x] Add global batch token per expert in wandb (#18)
3. - [x] Add MoE global batch loss metrics (#21)

## Others
1. - [x] Disable fused kernel building for ROCm (#8)
2. - [x] Remove redundant grad stats when --log-num-zeros-in-grad is not enabled (#9)
 
# Model Support

## Algorithm - @lx865712528 
1. - [x] Add cross entropy label smoothing (#16)
2. - [x] Support normal distribution initialization for output layers (#22)
3. - [x] Add Kaiming init option for MoE router weights (#25)

## MoE - @yzygitzh 
1. - [x] Add global batch load balancing loss (#7)
2. - [x] Support fine-grained recompute for MoE layer (#13)
3. - [x] Support gradient scale and normalization for MoE router (#17)
4. - [x] Add option to use different score function for aux loss (#26)

## Optimizer - @yzygitzh 
1. - [x] Fix the issue where the learning rate is not overridden when using the --override-opt_param-scheduler (#14)

## Tokenizer - @lx865712528 
1. - [x] Add option to allow trust_remote_code for HuggingFace Tokenizer (#23)

# Documentation & Repo

## Documentation
1. - [x] Add Microsoft SECURITY.MD (#2 and #5)
2. - [x] Add License (#4) 

## Repo 
1. - [x] Initiate code owners (#30)

# TODOs

# Framework Features

## Checkpoint - @abuccts
1. - [ ] Workaround hang issue when saving dist checkpoint in AMD GPUs (#28)
2. - [ ] Fix wrong offset in metadata of dist checkpoint (#29)

## Parallelism
1. - [ ] Enable interleaved-pp when layers in pp stage not divisible by vpp degree (#27)

## Repo 
1. - [ ] Build the CI/CD pipeline
